gits/terminal.py

```diff
@@ -589,48 +648,50 @@ def write(self, s):
                 self._echo(i)
 
     def dumphtml(self):
-        h = self._rows
-        w = self._cols
+        rows = self._rows
+        cols = self._cols
         r = ''
 
-        span = ''  # ready-to-output character
-        span_bg, span_fg = -1, -1
-        for i in range(h * w):
-            q, c = divmod(self._screen[i], 256 * 256 * 256)
+        span = ''  # ready-to-output characters
+        span_classes = []
+        for i in range(rows * cols):
+            q, c = divmod(self._screen[i], MAGIC_NUMBER)
             bg, fg = divmod(q, 16)
 
-            # Gits supports two color schemes: normal and bright. Each color
-            # scheme consists of 8 colors for a background and text. The
-            # terminal doesn't allow users to switch between them so far.
-            # Gits uses the normal color scheme by default.
-            #
-            # Suppose we have a bright green color (10). Using bitwise AND, we
-            # can get a normal green color (2).
-            bg &= 0x7
+            current_classes = [
+                'b{}'.format(bg),
+                'f{}'.format(fg)
+            ]
+
+            if self._screen[i] & (1 << UNDERLINE_BIT):
+                current_classes.append('underline')
+
+            if self._screen[i] & (1 << BLINK_BIT):
+                current_classes.append('blink')
+
+            if self._screen[i] & (1 << BOLD_BIT):
+                current_classes.append('bold')
 
-            if i == self._cur_y * w + self._cur_x:
-                bg, fg = 1, 7
+            if i == self._cur_y * cols + self._cur_x:
```

Comment:
Could you add more comments describing this if statement? Or simply extract it to the separate method with a human readable name.